### ISSUE TITLE
[Feature request] Add 'is_open' attribute to Issue resources

### ISSUE THREAD

**Issue by matletix** [maxtepkeev/python-redmine#281]
Currently, to find if an issue is open or closed, it is necessary to maintain a mapping between the possible `status_ids` and their  related open/closed status, or compute the difference between the issues returned by the `all()` method (by default only the open issues) and those returned by the `filter(status_id='*')` method (which also returns the closed issues).

Would it be possible to have this information directly available as an attribute of the `Issue` resources for example ? This way, if the statuses change, the mapping don't have to be updated and the correct information is directly available in the Issue's attributes.

Another place where this information could go is in the attribute of the `IssueStatus` object in addition to the `id` and `name` attributes.

**Comment by maxtepkeev** [comment]
We didn't implement this originally as the idea was that if someone needs that functionality it's quite straightforward to implement it by yourself, i.e. call issue_status.all(), put it into a dict and there's you mapping. But if that functionality is desired to be provided by Python-Redmine we can probably add an option to keep that mapping inside Python-Redmine, which will be off by default to not to make unnecessary call for those who don't need it.

**Comment by matletix** [comment]
Hi ! Thanks for your reply.

> call issue_status.all(), put it into a dict and there's you mapping.

Hmm, I might be missing something but when I do `dir(redmine.issue_status.all()[0])` I got `['id', 'issues', 'name']`, with `{'issues': None, 'id': 2, 'name': 'New'}`. I don't have the information on whether this issue status is open or closed.
So I have to maintain my own mapping with `{2: 'open'}` for example. Or did you mean something else ?

**Comment by maxtepkeev** [comment]
That is a desired behaviour, if you need to retrieve both open and closed issues use `redmine.issue.filter(project_id=ID, status_id='*')`

**Comment by matletix** [comment]
@maxtepkeev Ok thank you for the clarification.
And about the 1st part, do you know if there is a way to programmatically generate the mapping I was refering to in :
> So I have to maintain my own mapping with {2: 'open'} for example

i.e, a way to fetch the issue_status open/closed state programmatically

**Comment by maxtepkeev** [comment]
> So we could add the is_closed attribute to the IssueStatus object in python-redmine

We don't hardcode any attributes, so if ``is_closed`` is available in response you should see it in IssueStatus object as well, if you don't see, perhaps your Redmine instance doesn't return it in the response.

**Comment by matletix** [comment]
> We don't hardcode any attributes, so if is_closed is available in response you should see it in IssueStatus object as well

Oh ok, so I checked the `/issue_statuses.json` API endpoint and actually the `is_closed` attribute is returned, but ONLY with the issue statuses for which the `is_closed` attribute is `true`. Looking at the issue_status objects returned by python-redmine, I can indeed see the is_closed attribute for the issue_statuses which are considered closed.
This way, using `getattr(issue_status, 'is_closed', False)` on the issue_status objects allows me to dynamically create the desired mapping.
Closing this issue, thanks for your help !
